packages/discovery-react-components/src/utils/useDataApi.ts

```diff
@@ -172,6 +174,7 @@ interface ReducerState {
   data: any;
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   parameters: any;
+  error: string;
```

Comment:
i think we are going to want to handle the various cases from the `formatError` function in the SDK - https://github.com/IBM/node-sdk-core/blob/main/lib/request-wrapper.ts#L260-L323

axios has a lot of weird ways it produces errors, I believe that this function tries to normalize them. The actual error object seems to look something like
```typescript
interface SdkError {
  message: string;
  statusText: string;
  body?: Record<string, any>;
  headers?: Record<string, any>;
};
```
maybe we store the entire error object instead and rely on this type to get the structured data out of the response rather than trying to parse a string?

Reply:
This makes sense to me, thank you. Updated!